libraries/botbuilder-dialogs-adaptive-testing/src/httpRequestMocks/httpResponseMockMessage.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * @module botbuilder-dialogs-adaptive-testing
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { HttpResponseMessage, HttpResponseMock, ResponseStatusCode } from './httpResponseMock';
+import { HttpResponseMockContent } from './httpResponseMockContent';
+
+export class HttpResponseMockMessage {
+    private readonly _mock: HttpResponseMock;
+    private readonly _content: HttpResponseMockContent;
+
+    public constructor(httpResponseMock?: HttpResponseMock) {
+        if (httpResponseMock) {
+            this._mock = httpResponseMock;
+        } else {
+            this._mock = {
+                content: '',
+            };
+        }
+        this._content = new HttpResponseMockContent(this._mock);
+    }
+
+    public getMessage(): HttpResponseMessage {
+        let statusCode = ResponseStatusCode.OK;
+        if (typeof this._mock.statusCode === 'number') {
+            statusCode = this._mock.statusCode;
+        } else if (typeof this._mock.statusCode === 'string') {
+            statusCode = (ResponseStatusCode as unknown)[this._mock.statusCode] ?? ResponseStatusCode.OK;
```

Comment:
I think this might work natively: https://www.typescriptlang.org/docs/handbook/enums.html#reverse-mappings without the `as unknown` piece.

Reply:
unfortunately my enum value is a number, there is an error been thrown out with `as unknown` cast:
> Type 'string' is not assignable to type 'ResponseStatusCode'.